app/screens/thread/thread_follow_button/index.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
+import withObservables from '@nozbe/with-observables';
+
+import {MM_TABLES} from '@constants/database';
+
+import ThreadFollowButton from './thread_follow_button';
+
+import type {WithDatabaseArgs} from '@typings/database/database';
+import type ThreadModel from '@typings/database/models/servers/thread';
+
+const {SERVER: {THREAD}} = MM_TABLES;
+
+const enhanced = withObservables(['threadId'], ({threadId, database}: {threadId: string} & WithDatabaseArgs) => {
+    return {
+        thread: database.get<ThreadModel>(THREAD).findAndObserve(threadId),
```

Comment:
I believe you created a query or an observer for this

Reply:
Query was for fetching the thread by id but yes, we can have one for observe as well.

Reply:
hmmm I see observeThreadById right here https://github.com/mattermost/mattermost-mobile/pull/6075/files#diff-8aa2ba4bcdcb567c73fe742fe8aa74e1138f7a1e2002928941f4eb080976c1d6R45

Reply:
My bad, I renamed it and forgot it existed and got confused with the fetch one. I would've found it anyway while trying to write the same code again 😅